src/types/macros.ts

```diff
@@ -5,22 +5,25 @@ export interface MacroMap {
   [key: string]: string;
 }
 
+/**
+ * This function compares two MacroMap objects, and returns true if all properties
+ * are the same
+ * @param first first MacroMap object
+ * @param second second MacroMap object
+ * @returns boolean, true if they match else false
+ */
 export function macrosEqual(first: MacroMap, second: MacroMap): boolean {
-  // TODO why are the DIDs not matching?
   const { DID: firstDid, ...firstOthers } = first;
-  const { DID: secondDid, ...secondOthers } = first;
+  const { DID: secondDid, ...secondOthers } = second;
+
   for (const key in firstOthers) {
-    if (first.hasOwnProperty(key)) {
-      if (first[key] !== second[key]) {
-        return false;
-      }
+    if (first[key] !== second[key]) {
```

Comment:
Again, there's a reason why these lines of code were there, though arguable.

Reply:
My thinking was that it is iterating over the keys in firstOthers, and as firstOthers is a subset of first all keys in firstOthers are guaranteed to be in first (as long as the above bug where first is assigned to both lines doesn't happen again [which is now tested]). If there is another reason I am missing please let me know though.

Reply:
Oh I see. These were actually supposed to be `firstOthers.hasOwnProperty(key)`. It's all to do with a subtlety in how objects are compared in JavaScript. See https://stackoverflow.com/a/5859028 for an example.

However, we may well never hit the corner cases that is trying to allow for, so I'm not really inclined to change this back.